### PR TITLE
create webpack configuration

### PR DESCRIPTION
**Pull request by yurii-kokhalevych**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/browse/LVNOD-24)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-41)

## Code reviewers

- [x] @yarvanya
- [x] @denisbodnar
- [x] @shyyka

### Second Level Review

- [x] @lhalam
- [x] @4asof4ik

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo
## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=90%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": ["es2015"]
+}
```
Comment (denisbodnar) [inline .babelrc]: add a new line

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+    "extends": "standard",
+    "plugins": [
+        "standard",
+        "promise"
+    ]
+};
```
Comment (denisbodnar) [inline .eslintrc.js]: add a new line

File: webpack/webpack.config.dev.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+const path = require('path');
+const webpack = require('webpack');
+const nodeExternals = require('webpack-node-externals');
+
+
```
Comment (denisbodnar) [inline webpack/webpack.config.dev.js]: remove line

File: webpack/webpack.config.dev.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  new webpack.DefinePlugin({
+        'process.VERSION': require('package.json').version,
+        'process.browser': true
+      }),
+  new webpack.HotModuleReplacementPlugin(),
+  new webpack.NoEmitOnErrorsPlugin(),
+  ],
+
+  devtool: 'source-map'
+
```
Comment (denisbodnar) [inline webpack/webpack.config.dev.js]: remove line

**Review by denisbodnar — DISMISSED** [review]

**Review by denisbodnar — DISMISSED** [review]

**Review by yarvanya — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by denisbodnar — APPROVED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by yarvanya — DISMISSED** [review]

**Review by denisbodnar — DISMISSED** [review]

**Review by denisbodnar — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by yarvanya — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by denisbodnar — DISMISSED** [review]

**Review by yarvanya — DISMISSED** [review]

File: package.json [hunk 5]
```diff
@@ -7,7 +7,10 @@
     "test": "test"
   },
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "buid:prod": "webpack --config ./webpack/webpack.config.prod.js -p",
+    "dev": "webpack --config ./webpack/webpack.config.dev.js",
+    "start": "npm run dev"
```
Comment (4asof4ik) [inline package.json]: dev + start and rename start:dev

File: package.json [hunk 6]
```diff
@@ -7,7 +7,10 @@
     "test": "test"
   },
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "echo \"Error: no test specified\" && exit 1",
```
Comment (4asof4ik) [inline package.json]: drop if command is not exist

File: package.json (hunk 5 above)
Comment (4asof4ik) [reply package.json]: add production command

File: package.json [hunk 7]
```diff
@@ -22,10 +25,26 @@
...
+    "babel-loader": "^7.0.0",
+    "babel-preset-env": "^1.5.0",
+    "babel-preset-es2015": "^6.24.1",
+    "eslint": "^3.19.0",
+    "eslint-config-standard": "^10.2.1",
+    "eslint-plugin-promise": "^3.5.0",
+    "eslint-plugin-standard": "^3.0.1",
+    "nodemon": "^1.11.0",
+    "webpack": "^2.5.1"
   }
```
Comment (4asof4ik) [inline package.json]: 39-46 and 48 should be in depen..... ( today I explained difference between devDep... and dep )

File: webpack/webpack.config.dev.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    filename: 'bundle.js',
+  },
+  target: 'node',
+  externals: [nodeExternals()],
+
+  module: {
+    rules: [
+      {
+        test: /\.js?$/,
+        exclude: /node_modules/,
```
Comment (4asof4ik) [inline webpack/webpack.config.dev.js]: Why you use 'webpack-node-externals' if you excluded node_modules manual?

File: webpack/webpack.config.prod.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+  test: [/\.js$/],
+  exclude: /node_modules/,
+  loader: WebpackStripLoader.loader('console.log'),
+};
+
+devConfig.module.rules.push(stripLoader);
+
+module.exports = {
+  plugins: [
+    new CleanWebpackPlugin(['build'], {
```
Comment (4asof4ik) [inline webpack/webpack.config.prod.js]: create rootPath and buildPath var

File: webpack/webpack.config.prod.js [hunk 10]
```diff
@@ -0,0 +1,23 @@
+const WebpackStripLoader = require('strip-loader');
+let devConfig = require('./webpack.config.dev.js');
+const CleanWebpackPlugin = require('clean-webpack-plugin');
+
+const stripLoader = {
+  test: [/\.js$/],
+  exclude: /node_modules/,
+  loader: WebpackStripLoader.loader('console.log'),
```
Comment (4asof4ik) [inline webpack/webpack.config.prod.js]: webpack 2 'use' but no 'loader'

**Review by shyyka — DISMISSED** [review]

**Review by yarvanya — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by yarvanya — DISMISSED** [review]

**Review by denisbodnar — DISMISSED** [review]

File: package.json [hunk 11]
```diff
@@ -7,7 +7,9 @@
     "test": "test"
   },
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "build:prod": "NODE_ENV=production webpack -p --config ./webpack/webpack.config.prod.js",
+    "start:dev": "webpack --config ./webpack/webpack.config.dev.js",
```
Comment (4asof4ik) [inline package.json]: use webpack-dev-server

File: webpack/webpack.config.prod.js [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+  exclude: /node_modules/,
+  loader: 'strip-loader',
+};
+
+devConfig.module.rules.push(stripLoader);
+
+module.exports = {
+  plugins: [
+    new CleanWebpackPlugin(['buildPath'], {
+      rootPath: 'rootPath',
```
Comment (4asof4ik) [inline webpack/webpack.config.prod.js]: without ' '

**Review by yarvanya — DISMISSED** [review]

**Review by denisbodnar — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by denisbodnar — DISMISSED** [review]

**Review by yarvanya — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by denisbodnar — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by denisbodnar — APPROVED** [review]

**Review by shyyka — APPROVED** [review]

**Review by lhalam — APPROVED** [review]
good

**Review by yarvanya — APPROVED** [review]

File: package.json [hunk 13]
```diff
@@ -2,26 +2,10 @@
   "name": "api.da-mi",
   "version": "1.0.0",
   "description": "Back End",
-  "main": "app.js",
+  "main": ".eslintrc.js",
```
Comment (4asof4ik) [inline package.json]: why?

File: webpack/webpack.config.prod.js [hunk 14]
```diff
@@ -1,22 +1,22 @@
...
 const CleanWebpackPlugin = require('clean-webpack-plugin');
 const rootPath = __dirname + '/';
 const buildPath = __dirname + '/build';
+let devConfig = require('./webpack.config.dev.js');
 
 const stripLoader = {
   test: [/\.js$/],
   exclude: /node_modules/,
-  loader: 'strip-loader',
+  use: 'strip-loader',
```
Comment (4asof4ik) [inline webpack/webpack.config.prod.js]: this loader is deprecated

**Review by 4asof4ik — APPROVED** [review]

**Pull request merged**